### ISSUE TITLE
Would like to see a option to put g-code viewer on the Dashboard just like webcam

### ISSUE THREAD

**Issue by karlalbury** [mainsail-crew/mainsail#870]
### Requested feature:

Add option to pull the Gcode Viewer Tracking window to the Dashboard.
find my self bouncing between the two a lot.

Include Render of what looking for.
![mainsail](https://user-images.githubusercontent.com/5194171/172127362-5afc1371-5cd4-4668-8c38-c458ea234ac5.jpg)

**Comment by bnieves1994** [comment]
I would also love to see this.

**Comment by piwi3910** [comment]
I would also like to see this, The feature exists in Fluidd, something similar in mainsail would be great

**Comment by Sindarius** [comment]
I recommended against this because the 3D viewer is much more resource intensive compared to a 2d viewer. I think an alternative would be to bring some of the panels like progress and temps into the viewer page.

**Comment by Nazgile94** [comment]
> I recommended against this because the 3D viewer is much more resource intensive compared to a 2d viewer. I think an alternative would be to bring some of the panels like progress and temps into the viewer page.

its optional, if you dont like it - dont add the widget to the dashboard

i +1 this, would be nice ! maybe with toggle for 3d/2d

**Comment by PRGeno** [comment]
I agree with adding a 2D version of the GCode viewer to the dashboard especially for making Excluding Objects more useful and accessible.  I don't think the 3D viewer makes much sense on the dash, but for me a 2D viewer like Fluidd is a critical feature and one of the only reasons I typically use Fluidd over Mainsail.  

I think having the bed mesh on the dash like Fluidd would be nice to have too, but no where near as critical as a 2D GCode viewer.

**Comment by Jeppedy** [comment]
I'd be happy with the OctoPrint-style GCode Viewer.  One layer at a time, showing what will be laid down for that layer, and allow scrubbing to see future/past layers.  A nice simple 2D viewer compared to what Mainsail has right now.

**Comment by dot-Justin** [comment]
+1 this, would love to see.

**Comment by sjoerd104** [comment]
+1 would be great, even octo's 2d one.

**Comment by notwaffles** [comment]
+1

**Comment by marrrkus** [comment]
+1

**Comment by snuguru-maestro** [comment]
+1

**Comment by PeterXQC** [comment]
+1
